library/std/src/sys/hermit/futex.rs

```diff
@@ -16,7 +16,7 @@ pub fn futex_wait(futex: &AtomicU32, expected: u32, timeout: Option<Duration>) -
 
     let r = unsafe {
         abi::futex_wait(
-            futex.as_mut_ptr(),
+            futex.as_ptr() as *mut _,
```

Comment:
The returned pointer is still mutable, so this does not need casting.
```suggestion
            futex.as_ptr(),
```

Reply:
You are right. I fixed it.